src/app/shared/components/immediate-filter/immediate-filter.component.html

```diff
@@ -0,0 +1,6 @@
+<mat-form-field class="filter-field ml-3">
+  <label>
+    <input matInput (keyup)="onKeyUp($event, $event.target.value)"
```

Comment:
Even though this is a working solution, I would prefer using a FormControl. You will just simply create a FormControl object, pass it to the `input` element and subscribe to valueChanges.
```suggestion
    <input matInput [formControl]="formControl"
```
For more information: https://angular.io/guide/reactive-forms